sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -318,10 +320,9 @@ object Cast extends QueryErrorsBase {
     case _ if from == to => true
     case (NullType, _) => true
     case (_: NumericType, _: NumericType) => true
-    case (_: AtomicType, CharType(_) | VarcharType(_)) => false
-    case (_: CalendarIntervalType, CharType(_) | VarcharType(_)) => false
-    case (_: AtomicType, _: StringType) => true
-    case (_: CalendarIntervalType, _: StringType) => true
+    case (_: StringType, _: StringType) => true
+    case (_: AtomicType, s: StringType) if StringHelper.isPlainString(s) => true
```

Comment:
I'm wondering if it's better and more efficient to allow casting from atomic type to any string type. Then we don't need two back to back casts for it.